### ISSUE TITLE
java.lang.ArrayIndexOutOfBoundsException while running gds.beta.node2vec.stream

### ISSUE THREAD

**Issue by kinpoon-sn** [neo4j/graph-data-science#136]
**Describe the bug**

**To Reproduce**

GDS version: 1.7.2
Neo4j version: 4.3.4 Community
Operating system: Amazon Linux rhel fedora

- [ ] import data by `neo4j-admin import`, about 60M nodes, 700M edges
- [ ] query: 
```
CALL gds.beta.node2vec.stream(
          {
            nodeProjection: "*",
            relationshipProjection: "R",
            embeddingDimension: 128,
            randomSeed: 42,
            concurrency: 4,
            walkLength: 10,
            iterations: 1,
            relationshipProperties: ["weight"],
            relationshipWeightProperty: "weight"
          }
        )
        YIELD nodeId, embedding
        RETURN gds.util.asNode(nodeId).nodeId AS name, embedding, labels(gds.util.asNode(nodeId)) AS labels

```
error log (can run it successfully by fastRP)
```
2021-11-16 06:48:11.202+0000 INFO  [neo4j.BoltWorker-4 [bolt] [/127.0.0.1:43886] ] Node2Vec :: RandomWalk :: create walks 100%
2021-11-16 06:48:11.203+0000 INFO  [neo4j.BoltWorker-4 [bolt] [/127.0.0.1:43886] ] Node2Vec :: RandomWalk :: create walks :: Finished
2021-11-16 06:48:11.203+0000 INFO  [neo4j.BoltWorker-4 [bolt] [/127.0.0.1:43886] ] Node2Vec :: RandomWalk :: Finished
2021-11-16 06:55:35.822+0000 INFO  [neo4j.BoltWorker-4 [bolt] [/127.0.0.1:43886] ] Node2Vec :: train :: Start
2021-11-16 06:55:35.824+0000 INFO  [neo4j.BoltWorker-4 [bolt] [/127.0.0.1:43886] ] Node2Vec :: train :: iteration 1 of 1 :: Start
2021-11-16 06:55:42.614+0000 WARN  Computation failed
java.lang.ArrayIndexOutOfBoundsException: Index 8 out of bounds for length 8
        at org.neo4j.gds.core.loading.ZigZagLongDecoding.zigZagUncompress(ZigZagLongDecoding.java:32) ~[neo4j-graph-data-science-1.7.2.jar:?]
        at org.neo4j.gds.core.loading.ZigZagLongDecoding.zigZagUncompress(ZigZagLongDecoding.java:25) ~[neo4j-graph-data-science-1.7.2.jar:?]
        at org.neo4j.gds.embeddings.node2vec.CompressedRandomWalks$CompressedWalkIterator.next(CompressedRandomWalks.java:141) ~[neo4j-graph-data-science-1.7.2.jar:?]
        at org.neo4j.gds.embeddings.node2vec.CompressedRandomWalks$CompressedWalkIterator.next(CompressedRandomWalks.java:95) ~[neo4j-graph-data-science-1.7.2.jar:?]
        at org.neo4j.gds.embeddings.node2vec.PositiveSampleProducer.nextWalk(PositiveSampleProducer.java:78) ~[neo4j-graph-data-science-1.7.2.jar:?]
        at org.neo4j.gds.embeddings.node2vec.PositiveSampleProducer.nextCenterWord(PositiveSampleProducer.java:100) ~[neo4j-graph-data-science-1.7.2.jar:?]
        at org.neo4j.gds.embeddings.node2vec.PositiveSampleProducer.nextContextWord(PositiveSampleProducer.java:122) ~[neo4j-graph-data-science-1.7.2.jar:?]
        at org.neo4j.gds.embeddings.node2vec.PositiveSampleProducer.nextContextWord(PositiveSampleProducer.java:125) ~[neo4j-graph-data-science-1.7.2.jar:?]
        at org.neo4j.gds.embeddings.node2vec.PositiveSampleProducer.nextContextWord(PositiveSampleProducer.java:125) ~[neo4j-graph-data-science-1.7.2.jar:?]
        at org.neo4j.gds.embeddings.node2vec.PositiveSampleProducer.next(PositiveSampleProducer.java:65) ~[neo4j-graph-data-science-1.7.2.jar:?]
        at org.neo4j.gds.embeddings.node2vec.Node2VecModel$TrainingTask.run(Node2VecModel.java:189) ~[neo4j-graph-data-science-1.7.2.jar:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) ~[?:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[?:?]
        at java.lang.Thread.run(Thread.java:829) [?:?]
        at org.neo4j.internal.helpers.NamedThreadFactory$2.run(NamedThreadFactory.java:110) ~[neo4j-common-4.3.4.jar:4.3.4]
        Suppressed: java.lang.ArrayIndexOutOfBoundsException: Index 8 out of bounds for length 8
```

**Comment by soerenreichardt** [comment]
Hello @kinpoon-sn,
thanks for the bug report! We were able to locate and fix the issue. The fix should be available in all upcoming patch releases of GDS 1.7 and 1.8.

**Comment by kinpoon-sn** [comment]
thank you @soerenreichardt , looking forward to the patch
